Sources/ZcashLightClientKit/Transaction/WalletTransactionEncoder.swift

```
@@ -54,93 +54,52 @@ class WalletTransactionEncoder: TransactionEncoder {
         shieldingThreshold: Zatoshi,
-        memoBytes: MemoBytes?,
-        from accountIndex: Int
-    ) async throws -> ZcashTransaction.Overview {
-        let txId = try await createShieldingSpend(
-            spendingKey: spendingKey,
-            shieldingThreshold: shieldingThreshold,
+        memoBytes: MemoBytes?
+    ) async throws -> Proposal {
+        let proposal = try await rustBackend.proposeShielding(
+            account: Int32(accountIndex),
             memo: memoBytes,
-            accountIndex: accountIndex
+            shieldingThreshold: shieldingThreshold
         )
-        
-        logger.debug("transaction id: \(txId)")
-        return try await repository.find(rawID: txId)
+
+        return Proposal(inner: proposal)
     }
 
-    func createShieldingSpend(
-        spendingKey: UnifiedSpendingKey,
-        shieldingThreshold: Zatoshi,
-        memo: MemoBytes?,
-        accountIndex: Int
-    ) async throws -> Data {
+    func createProposedTransactions(
+        proposal: Proposal,
+        spendingKey: UnifiedSpendingKey
+    ) async throws -> [ZcashTransaction.Overview] {
         guard ensureParams(spend: self.spendParamsURL, output: self.outputParamsURL) else {
-            throw ZcashError.walletTransEncoderShieldFundsMissingSaplingParams
+            throw ZcashError.walletTransEncoderCreateTransactionMissingSaplingParams
         }
 
-        // TODO: Expose the proposal in a way that enables querying its fee.
-        let proposal = try await rustBackend.proposeShielding(
-            account: Int32(spendingKey.account),
-            memo: memo,
-            shieldingThreshold: shieldingThreshold
-        )
-
         let txId = try await rustBackend.createProposedTransaction(
-            proposal: proposal,
+            proposal: proposal.inner,
             usk: spendingKey
         )
 
-        return txId
+        logger.debug("transaction id: \(txId)")
+        return [try await repository.find(rawID: txId)]
```

Does `repository.find` throw if it cannot find the txid? I would think that ideally that would return an optional result, so that "not found" is distinguished from "error" (though in this particular case you'd want to throw an error anyway, so it's nonblocking for now.)

---

We shouldn't encounter this case (successful wallet DB lookup that returns no result) because if we reach here, the transaction was successfully created by the Rust backend (and didn't itself result in an exception). Any exception thrown here would instead be due to database access failures, which are indeed exceptions.